lib/experimental/media/class-gutenberg-rest-attachments-controller.php

```diff
@@ -0,0 +1,350 @@
+<?php
+/**
+ * Class Gutenberg_REST_Attachments_Controller.
+ *
+ * @package MediaExperiments
+ */
+
+/**
+ * Class Gutenberg_REST_Attachments_Controller.
+ */
+class Gutenberg_REST_Attachments_Controller extends WP_REST_Attachments_Controller {
+	/**
+	 * Registers the routes for attachments.
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes(): void {
+		parent::register_routes();
+
+		$args                       = $this->get_endpoint_args_for_item_schema();
+		$args['generate_sub_sizes'] = array(
+			'type'        => 'boolean',
+			'default'     => true,
+			'description' => __( 'Whether to generate image sub sizes.', 'gutenberg' ),
+		);
+		$args['convert_format']     = array(
+			'type'        => 'boolean',
+			'default'     => true,
+			'description' => __( 'Whether to support server-side format conversion.', 'gutenberg' ),
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'create_item' ),
+					'permission_callback' => array( $this, 'create_item_permissions_check' ),
+					'args'                => $args,
+				),
+				'allow_batch' => $this->allow_batch,
+				'schema'      => array( $this, 'get_public_item_schema' ),
+			),
+			true
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<id>[\d]+)/sideload',
```

Comment:
Is `sideload` a good name for this? I've personally never fully understood that naming in core either, but why not make this more clear in what it does? Maybe `image_size`? Or `upload_image_size`?

Reply:
Well, I think we already have a better one, so why not go with that? `sideload` isn't clear about what this endpoint actually does. :)

Reply:
I don't think `image_size` or `upload_image_size` is better though

Reply:
Also, in the future this could be expanded to more than just adding/replacing image sizes, so the proposed names are too limiting IMO.

Reply:
Why not? More importantly, what does `sideload` even mean?

Reply:
As I said, it‘s too limiting. Something like `partial` would be better perhaps.

I don‘t really want to hold up the rest of the work by this naming question though. I recommend looking at the full code in context of https://github.com/swissspidy/media-experiments and open an issue there with a proposed name change, as it has implications for a lot of the JS code too and this way you see the full picture. Once that‘s settled over there, it‘s easier to port this over.